[PyTorch] Guard against self assignment in SymInt

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #84375

self assignment was broken, now it's not.

Differential Revision: [D39189342](https://our.internmc.facebook.com/intern/diff/D39189342/)

swolchok committed:
[PyTorch] Guard against self assignment in SymInt

self assignment was broken, now it's not.

Differential Revision: [D39189342](https://our.internmc.facebook.com/intern/diff/D39189342/)

[ghstack-poisoned]

suo reviewed (approved):
change looks good, although the tested self assignment is making CI sad

swolchok committed:
suppress another warning for PyTorch CI on "[PyTorch] Guard against self assignment in SymInt"


self assignment was broken, now it's not.

Differential Revision: [D39189342](https://our.internmc.facebook.com/intern/diff/D39189342/)

[ghstack-poisoned]

swolchok commented:
@pytorchbot merge -g

swolchok committed:
give up testing self-move-assignment on "[PyTorch] Guard against self assignment in SymInt"


self assignment was broken, now it's not.

Differential Revision: [D39189342](https://our.internmc.facebook.com/intern/diff/D39189342/)

[ghstack-poisoned]

swolchok commented:
@pytorchbot merge -g

swolchok committed:
regrettably, axe the test on "[PyTorch] Guard against self assignment in SymInt"


self assignment was broken, now it's not.

Differential Revision: [D39189342](https://our.internmc.facebook.com/intern/diff/D39189342/)

[ghstack-poisoned]

swolchok commented:
@pytorchbot merge -g
